ts/interactions/conversationInteractions.ts

```diff
@@ -225,54 +237,186 @@ export async function showUpdateGroupMembersByConvoId(conversationId: string) {
+        status: ConversationInteractionStatus.Start,
+      });
+      onClickClose();
+      await getConversationController().delete1o1(conversationId, {
+        fromSyncMessage: false,
+        justHidePrivate: true,
+        keepMessages: isMe,
+      });
+      await clearConversationInteractionState({ conversationId });
+    } catch (err) {
+      window.log.warn(`showLeavePrivateConversationbyConvoId error: ${err}`);
+      await saveConversationInteractionErrorAsMessage({
+        conversationId,
+        interactionType: isMe
+          ? ConversationInteractionType.Hide
+          : ConversationInteractionType.Leave,
+      });
+    }
+  };
+
+  window?.inboxStore?.dispatch(
+    updateConfirmModal({
+      title: isMe ? window.i18n('hideConversation') : window.i18n('deleteConversation'),
+      message: isMe
+        ? window.i18n('hideNoteToSelfConfirmation')
+        : window.i18n('deleteConversationConfirmation', name ? [name] : ['']),
+      onClickOk,
+      okText: isMe ? window.i18n('hide') : window.i18n('delete'),
+      okTheme: SessionButtonColor.Danger,
+      onClickClose,
+      conversationId,
+    })
+  );
+}
+
+async function leaveGroupOrCommunityByConvoId(
+  conversationId: string,
+  isPublic: boolean,
+  forceDeleteLocal: boolean,
+  onClickClose?: () => void
+) {
+  try {
+    await updateConversationInteractionState({
+      conversationId,
+      type: ConversationInteractionType.Leave,
+      status: ConversationInteractionStatus.Start,
+    });
+
+    if (onClickClose) {
+      onClickClose();
+    }
+
+    if (isPublic) {
+      await getConversationController().deleteCommunity(conversationId, {
+        fromSyncMessage: false,
+      });
+    } else {
+      await getConversationController().deleteClosedGroup(conversationId, {
+        fromSyncMessage: false,
+        sendLeaveMessage: true,
+        forceDeleteLocal,
+      });
+    }
+    await clearConversationInteractionState({ conversationId });
+  } catch (err) {
+    window.log.warn(`showLeaveGroupByConvoId error: ${err}`);
+    await saveConversationInteractionErrorAsMessage({
+      conversationId,
+      interactionType: ConversationInteractionType.Leave,
+    });
+  }
+}
+
+export async function showLeaveGroupByConvoId(conversationId: string, name: string | undefined) {
   const conversation = getConversationController().get(conversationId);
 
   if (!conversation.isGroup()) {
     throw new Error('showLeaveGroupDialog() called with a non group convo.');
   }
 
-  const title = window.i18n('leaveGroup');
-  const message = window.i18n('leaveGroupConfirmation');
-  const isAdmin = (conversation.get('groupAdmins') || []).includes(
-    UserUtils.getOurPubKeyStrFromCache()
-  );
   const isClosedGroup = conversation.isClosedGroup() || false;
   const isPublic = conversation.isPublic() || false;
+  const admins = conversation.get('groupAdmins') || [];
+  const isAdmin = admins.includes(UserUtils.getOurPubKeyStrFromCache());
```

Comment:
there is a `useWeAreAdmin` hook for that if you want

Reply:
This actually brings up a good point I never thought of. Is it okay to use hooks anywhere in our codebase? In my mind I thought it should only be used as part of a react component?

Reply:
Yeah it doesn't look like you can. I got this error in the console. Assume it has something to do with the function being async?

```
Top-level unhandled promise rejection: Error: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```

Reply:
no, you're right. Those hooks can only be called on other hooks, or react components. In your case it's none of those so it doesn't work and crash with that 321 error which is specifically (invalid hook call). 

Also, in your case that's fine, as we only care about the isAdmin for closed group case, but if we cared about the isAdmin or moderator case for public group, we'd need to specifically do both tests.